### PR TITLE
provider: Add TimestreamWrite service client (#15421)

### PR DESCRIPTION
**Pull request by yardensachs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #15421

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by yardensachs** [comment]
This PR is 1 out of 2 PRs, split from #15444 per @DrFaust92 request

**Comment by ewbankkit** [comment]
@yardensachs A maintainer must merge https://github.com/terraform-providers/terraform-provider-aws/pull/15226 first to pull in the latest AWS SDK.

**Comment by bflad** [comment]
Updated AWS Go SDK dependency has been merged, so this should be happier when rebased. 👍

File: website/allowed-subcategories.txt [hunk 1]
```diff
@@ -110,6 +110,7 @@ SimpleDB
 Step Function (SFN)
 Storage Gateway
 Synthetics
+TimestreamWrite
```
Comment (bflad) [inline website/allowed-subcategories.txt:113]: Human facing category can have the space in the name. 😄 

```suggestion
Timestream Write
```

**Review by bflad — APPROVED** [review]
One nit, otherwise looks good once rebased. 🚀

File: website/allowed-subcategories.txt (hunk 1 above)
Comment (yardensachs) [reply website/allowed-subcategories.txt:113]: Will do!

**Comment by yardensachs** [comment]
asd

**Pull request merged**